baremetal/metal3machine_manager_test.go

```diff
@@ -2268,10 +2268,8 @@ var _ = Describe("Metal3Machine manager", func() {
 				))
 			}
 
-			// if we had to create an additional secret (dataSecretName not set and
-			// Data set)
-			if tc.Machine.Spec.Bootstrap.DataSecretName == nil &&
-				tc.Machine.Spec.Bootstrap.Data != nil {
+			// if we had to create an additional secret (dataSecretName not set)
+			if tc.Machine.Spec.Bootstrap.DataSecretName == nil {
 
 				Expect(tc.M3Machine.Status.UserData.Name).To(Equal(
 					tc.M3Machine.Name + "-user-data",
```

Comment:
you're not creating it anymore, no ? this might need a change, no ?

Reply:
I was removing `Bootstrap.Data` and related test cases since it was [deprecated](https://github.com/kubernetes-sigs/cluster-api/pull/4000) in CAPI, do we need to do something else instead?

Reply:
Also please check this [comment](https://github.com/metal3-io/cluster-api-provider-metal3/pull/167#discussion_r652824483)